gameobjects/ball.cpp

```diff
@@ -28,15 +33,15 @@ void Ball::OnStart()
 
 void Ball::OnUpdate()
 {
-    const glm::vec3 newPosition = GetPosition() + mVelocity;
+    const glm::vec3 newPosition = GetPosition() + mVelocity * Timer::frameTime;
     SetPosition(newPosition);
 }
 
 void Ball::OnCollisionStart(GameObject& other)
 {
     if (other.GetInstanceName() == "ScoreArea")
     {
-        ResetBall();
+        SetTimeout(BALL_RESET_WAIT_MS, std::bind(&Ball::ResetBall, this));
```

Comment:
std::bind has fallen somewhat out of fashion. I think you should be able to get away with a lambda that captures `this`. Worth a try to see if it works, not sure the impact when using it with std::function

Aka:

```
SetTimeout(BALL_RESET_WAIT_MS, [this] ()
{
    this->ResetBall();
}
```

or something like that

Reply:
It does work, very cool!